.github/workflows/ci.yaml

```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  lint:
+    name: Lint with `mix check`
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Install OTP and Elixir
+      uses: erlef/setup-beam@v1
+      id: beam
+      with:
+        version-file: .tool-versions
+        version-type: strict
+
+    - name: Cache deps and build
+      uses: actions/cache@v3
+      with:
+        path: |
+          deps
+          _build
+        key: "${{ runner.os }}-${{ steps.beam.outputs.otp-version }}-${{ steps.beam.outputs.elixir-version }}-${{ hashFiles('**/mix.lock') }}"
```

Comment:
You have two jobs (lint and test) that build and use a cache.

However, lint uses the dev environment, test uses the test environment.

This means that when those two race to submit a cache for the _build directory, each of them submits one that only contains the one for its own environment. This in turn means that the cache is invalid for at least one of the jobs.

I suggest using different cache keys for each job, e.g.:
- for lint, `"lint-${{ runner.os }}-${{ steps.beam.outputs.otp-version }}-${{ steps.beam.outputs.elixir-version }}-${{ hashFiles('**/mix.lock') }}"`
- for test, `"test-${{ runner.os }}-${{ steps.beam.outputs.otp-version }}-${{ steps.beam.outputs.elixir-version }}-${{ hashFiles('**/mix.lock') }}"`